src/main/java/org/candlepin/resource/OwnerContentResource.java

```diff
@@ -83,134 +97,227 @@ public OwnerContentResource(OwnerCurator ownerCurator,
+            .map(cdto -> this.createContentImpl(owner, cdto))
+            .map(this.translator.getStreamMapper(Content.class, ContentDTO.class));
     }
 
     @Override
-    public ContentDTO getOwnerContent(
-        @Verify(Owner.class) String ownerKey, String contentId) {
-
-        Owner owner = this.getOwnerByKey(ownerKey);
-        Content content = this.fetchContent(owner, contentId);
-
-        return this.translator.translate(content, ContentDTO.class);
+    public ContentDTO createContent(String ownerKey, ContentDTO content) {
+        return this.createContentBatch(ownerKey, List.of(content))
+            .findAny()
+            .orElse(null);
     }
 
     @Override
-    public CandlepinQuery<ContentDTO> listOwnerContent(@Verify(Owner.class) String ownerKey) {
+    @Transactional
+    public ContentDTO getContentById(@Verify(Owner.class) String ownerKey, String contentId) {
         Owner owner = this.getOwnerByKey(ownerKey);
-        CandlepinQuery<Content> query = this.ownerContentCurator.getContentByOwnerCPQ(owner);
+        Content content = this.resolveContentId(owner, contentId, null);
 
-        return this.translator.translateQuery(query, ContentDTO.class);
+        return this.translator.translate(content, ContentDTO.class);
     }
 
     @Override
     @Transactional
-    public Collection<ContentDTO> createBatchContent(String ownerKey, List<ContentDTO> contents) {
-        for (ContentDTO content : contents) {
-            this.validateContentForCreation(content);
-        }
+    public Stream<ContentDTO> getContentsByOwner(@Verify(Owner.class) String ownerKey,
+        List<String> contentIds, Boolean omitGlobalEntities) {
 
         Owner owner = this.getOwnerByKey(ownerKey);
+        String namespace = owner.getKey();
 
-        List<ContentDTO> output = contents.stream()
-            .map(content -> this.createContentImpl(owner, content))
-            .map(this.translator.getStreamMapper(Content.class, ContentDTO.class))
-            .collect(Collectors.toList());
+        Collection<Content> contents;
 
-        this.contentAccessManager.syncOwnerLastContentUpdate(owner);
+        if (omitGlobalEntities == null || !omitGlobalEntities) {
+            contents = contentIds != null && !contentIds.isEmpty() ?
+                this.contentCurator.resolveContentIds(namespace, contentIds).values() :
+                this.contentCurator.resolveContentsByNamespace(namespace);
+        }
+        else {
+            contents = contentIds != null && !contentIds.isEmpty() ?
+                this.contentCurator.getContentsByIds(namespace, contentIds).values() :
+                this.contentCurator.getContentsByNamespace(namespace);
+        }
+
+        Stream<ContentDTO> stream = contents.stream()
+            .map(this.translator.getStreamMapper(Content.class, ContentDTO.class));
 
-        return output;
+        return this.pagingUtilFactory.forClass(ContentDTO.class)
+            .applyPaging(stream, contents.size());
```

Comment:
This is the kind of endpoint paging at the query level would help: right now there are 12k+ contents in existence, and we know some of those massive orgs have 8-9k of them. This endpoint would fetch all of them in memory, only to apply paging after the fact

Reply:
True enough, and a good warning; but with this refactor, those content numbers will get paired down to ~12k total.